### PR TITLE
Adding util(process) module

### PR DESCRIPTION
**Pull request by goravsingal**
Adding util(process) module

### CODE REVIEW

File: hubblestack/extmods/hubble_mods/util.py [hunk 1]
```diff
@@ -0,0 +1,977 @@
...
+- encode_base64
+    Given a string, base64 encode it and return it.
+
+FDG Profile Example for one utility method:
+
+main:
+  module: util
+    args:
+        starting_seq: [3, 4]
+        filter_rules:
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py:160]: no explanation in documentation about this parameter

File: hubblestack/extmods/hubble_mods/util.py [hunk 2]
```diff
@@ -0,0 +1,977 @@
+# -*- encoding: utf-8 -*-
+"""
+Module having utility methods for data processing
+=================================================
+
+This module primarily processes and properly format
+the data outputted by a module to serve it to another module.
+
+All functions in this module are meant to be used in chaining.
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py:9]: Does this mean that we should block all calls from audit module ?

File: hubblestack/extmods/hubble_mods/util.py [hunk 3]
```diff
@@ -0,0 +1,977 @@
...
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': "Output (can be string/dict/list)", 'status': True},
+                  'caller': 'Audit'}
+
+    Raises:
+        AuditCheckValidationError: For any validation error
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py]: Typo in error name

File: hubblestack/extmods/hubble_mods/util.py [hunk 4]
```diff
@@ -0,0 +1,977 @@
...
+    """
+    log.debug('Module: service Start validating params for check-id: {0}'.format(block_id))
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py]: typo in module name

File: hubblestack/extmods/hubble_mods/util.py [hunk 5]
```diff
@@ -0,0 +1,977 @@
...
+    This function just returns the chained value. It is a nop/no operation.
+- encode_base64
+    Given a string, base64 encode it and return it.
+
+FDG Profile Example for one utility method:
+
+main:
+  module: util
+    args:
+        starting_seq: [3, 4]
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py]: function param is not provided in this block which is mandatory param

File: hubblestack/extmods/hubble_mods/util.py [hunk 6]
```diff
@@ -0,0 +1,977 @@
...
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': "Output (can be string/dict/list)", 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing stat module for id: {0}'.format(block_id))
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py]: typo in module name

File: hubblestack/extmods/hubble_mods/util.py [hunk 7]
```diff
@@ -0,0 +1,977 @@
...
+        elif function_param == ['dict_convert_none', 'print_string', 'dict_to_list', 
+            'get_index', 'dict_remove_none', 'nop', 'encode_base64']:
+            # no mandatory key, mentioned here just for clarity
+            pass
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py:234]: please add get_filtered_params_to_log function in this module

File: hubblestack/extmods/hubble_mods/util.py [hunk 8]
```diff
@@ -0,0 +1,977 @@
...
+        return val1 == val2
+    if comp == "ne":
+        return val1 != val2
+
+    log.error("Invalid argument '%s' - should be in [gt, ge, lt, le, eq, ne]", comp)
+    raise ArgumentValueError
+
+def _get_index(block_id, block_dict, extra_args):
+    """
+    Given a list list, return the item found at ``index``.
```
Comment (devagupt) [inline hubblestack/extmods/hubble_mods/util.py:455]: typo fix

**Review by devagupt — CHANGES_REQUESTED** [review]

File: hubblestack/extmods/hubble_mods/util.py (hunk 1 above)
Comment (goravsingal) [reply hubblestack/extmods/hubble_mods/util.py:160]: Added detailed doc

**Review by devagupt — APPROVED** [review]

**Pull request merged**